logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/UpdateApiVersionsWorker.kt

```diff
@@ -1,32 +1,17 @@
 package com.wire.kalium.logic.sync
 
-import com.wire.kalium.logic.CoreFailure
-import com.wire.kalium.logic.feature.server_config.UpdateApiVersionsResult
-import com.wire.kalium.logic.feature.server_config.UpdateApiVersionsUseCase
-import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.logic.feature.server.UpdateApiVersionsUseCase
 
 
 class UpdateApiVersionsWorker(
     private val apiVersionCheckManager: ApiVersionCheckManager,
-    private  val updateApiVersionsUseCase: UpdateApiVersionsUseCase
+    private val updateApiVersionsUseCase: UpdateApiVersionsUseCase
 ) : DefaultWorker() {
 
     override suspend fun doWork(): Result {
         apiVersionCheckManager.changeState(ApiVersionCheckState.Running)
-        return when (val result = updateApiVersionsUseCase()) {
-            is UpdateApiVersionsResult.Success -> {
-                kaliumLogger.e("UPDATE API VERSIONS SUCCESS ${result.serverConfigList}")
-                apiVersionCheckManager.changeState(ApiVersionCheckState.Completed)
-                Result.Success
-            }
-            is UpdateApiVersionsResult.Failure -> {
-                kaliumLogger.e("UPDATE API VERSIONS FAILURE ${result.genericFailure}")
-                (result.genericFailure as? CoreFailure.Unknown)?.let {
-                    it.rootCause?.printStackTrace()
-                }
-                apiVersionCheckManager.changeState(ApiVersionCheckState.Failed(result.genericFailure))
-                Result.Retry // TODO should it retry or just fail?
-            }
+        return updateApiVersionsUseCase().let { Result.Success }.also {
```

Comment:
What if we want to schedule the immediate work to be started when the app is launched just like the slow sync? Shouldn't we handle somehow the error and block the user from using the app without calculated common api version? Will such case be handled differently?

Reply:
I was thinking about this case and that should be handled for each account separately because the update can success for account A put fail for account B.
we can either handle this case in the sync after (at the beginning of the sync, check the API version).
Or we can return a list of failed update attempts (this way)